java/src/com/ibm/streamsx/topology/spl/SPL.java

```diff
@@ -27,34 +31,105 @@
+ * When necessary use {@code createParamValue(T, MetaType)}
+ * to create parameter values for SPL types.
  * For example:
  * <pre>{@code
  * Map<String,Object> params = ...
- * params.put("aUStringParam", new UString(...));
- * params.put("aUShortParam", new UnsignedShort(13));
+ * params.put("aInt32", 13);
+ * params.put("aUInt32", SPL.createParamValue(13, MetaType.UINT32));
+ * params.put("aRString", "some string value");
+ * params.put("aUString", SPL.createParamValue("some ustring value", MetaType.USTRING));
  * ... = SPLPrimitive.invokeOperator(..., params);
  * }</pre>
  * <p>
  * In addition to the usual Java types used for operator parameter values,
  * a {@code Supplier<T>} parameter value may be specified.
  * Submission time parameters are passed in this manner.
- * See {@link Topology#createSubmissionParameter(String, Class)}.
+ * See {@link #createSubmissionParameter(Topology, String, Object, boolean)}.
  * For example:
  * <pre>{@code
  * Map<String,Object> params = ...
- * params.put("aLongParam", topology.createSubmissionParameter(..., Long.class);
- * params.put("aShortParam", topology.createSubmissionParameter(..., (Short)13);
- * params.put("aULongParam", topology.createSubmissionParameter(..., UnsignedLong.class);
- * params.put("aUShortParam", topology.createSubmissionParameter(..., new UnsignedShort(13));
- * params.put("aUStringParam", topology.createSubmissionParameter(..., new UString(...));
+ * params.put("aInt32", topology.createSubmissionParameter("int32Param", 13);
+ * params.put("aUInt32", SPL.createSubmissionParameter(topology, "uint32Param", SPL.createParamValue(13, MetaType.UINT32), true);
+ * params.put("aRString", topology.createSubmissionParameter("rstrParam", "some string value");
+ * params.put("aUString", SPL.createSubmissionParameter(topology, "ustrParam", SPL.createParamValue("some ustring value", MetaType.USTRING), true);
+ * params.put("aUInt8", SPL.createSubmissionParameter(topology, "uint8Param", SPL.createParamValue((byte)13, MetaType.UINT8), true);
  * ... = SPLPrimitive.invokeOperator(..., params);
  * }</pre>
  */
 public class SPL {
+    
+    /**
+     * Create a SPL parameter value wrapper object for the 
+     * specified SPL {@code MetaType}.
+     * <p>
+     * Use of this is required to construct a SPL operator parameter value
+     * whose SPL type is not implied from simple Java type.  e.g.,
+     * a {@code String} value is interpreted as a SPL {@code rstring},
+     * and {@code Byte,Short,Integer,Long} are interpreted as SPL signed integers.
+     * Hence, this must be used for SPL {@code ustring} and unsigned integers.
+     * @param value the value to wrap
+     * @param metaType the SPL meta type
+     * @return the wrapper object
+     * @throws IllegalArgumentException if value is null or it's class is
+     *     not appropriate for {@code metaType}
+     * @see #paramValueToString(Object)
+     */
+    public static <T> Object createParamValue(T value, MetaType metaType) {
```

Comment:
Just wondering if this is simply `createValue`, could it in the future be used elsewhere, not as a parameter?

(also it's above should be its)